server/rest/src/main/java/org/infinispan/rest/resources/SecurityResource.java

```diff
@@ -152,6 +165,57 @@ private CompletionStage<RestResponse> describeRole(RestRequest request) {
       return asJsonResponseFuture(invocationHelper.newResponse(request), json , isPretty(request));
    }
 
+   private CompletionStage<RestResponse> listAllPrincipals(RestRequest request, boolean detailed) {
+      if (!detailed) {
+         return asJsonResponseFuture(invocationHelper.newResponse(request),
+               Json.make(principalRoleMapper.listPrincipals()),
+               isPretty(request));
+      }
+
+      Set<Map.Entry<String, ClusterRoleMapper.RoleSet>> entries = principalRoleMapper.listPrincipalsAndRoleSet();
+      Json json = Json.object();
+      entries.stream().forEach(e -> json.set(e.getKey(), e.getValue().getRoles()));
+      return asJsonResponseFuture(invocationHelper.newResponse(request),
+            json,
+            isPretty(request));
+   }
+
+
+   private CompletionStage<RestResponse> describeRoleDetailed(RestRequest request) {
+      NettyRestResponse.Builder builder = invocationHelper.newResponse(request);
+      if (rolePermissionMapper == null) {
+         return completedFuture(invocationHelper.newResponse(request).status(CONFLICT).entity(Log.REST.rolePermissionMapperNotMutable()).build());
+      }
+      String roleName = request.variables().get("role");
+
+      GlobalAuthorizationConfiguration authorization = invocationHelper.getRestCacheManager().getInstance().getCacheManagerConfiguration().security().authorization();
+      Role role = authorization.getRole(roleName);
+
+      if (role == null) {
+         return completedFuture(builder.status(NOT_FOUND).build());
+      }
+
+      Json json = new RoleJson(role).toJson();
+      json.set("name", roleName);
+
+      // secured caches
+      Set<String> caches = invocationHelper.getRestCacheManager().getAccessibleCacheNames().stream().map(cacheName -> {
```

Comment:
@tristantarrant I implented this in it's own rest endpoint 
https://github.com/infinispan/infinispan/pull/11415